compiler/rustc_middle/src/ty/structural_impls.rs

```diff
@@ -343,14 +344,27 @@ impl<'tcx> DebugWithInfcx<TyCtxt<'tcx>> for ty::Const<'tcx> {
         this: OptWithInfcx<'_, TyCtxt<'tcx>, InfCtx, &Self>,
         f: &mut core::fmt::Formatter<'_>,
     ) -> core::fmt::Result {
-        // This reflects what `Const` looked liked before `Interned` was
-        // introduced. We print it like this to avoid having to update expected
-        // output in a lot of tests.
+        // If this is a value, we spend some effort to make it look nice.
+        if let ConstKind::Value(_) = this.data.kind() {
+            return ty::tls::with(move |tcx| {
+                // Somehow trying to lift the valtree results in lifetime errors, so we lift the
+                // entire constant.
```

Comment:
Turns out the Valtree Lift impl [is bogus](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/.60tcx.2Elift.60.20on.20valtrees/near/391348337).

Reply:
Is it used? #115831 is trying to remove a bunch of unused `Lift` and `Valtree` is one of them.

Reply:
Note: #115831 is (coincidentally) about to remove a bunch of unused `Lift` impls, including the bogus one for `Valtree`. This shouldn't cause you problems given that you'll need to adjust the implementation, but I'm mentioning it just so it doesn't catch you unawares.

Reply:
Ah, yeah not having it seems better than having a broken one. :) But it does mean the comment here will be outdated. I can't update it any more since the PR was already rolled up.

I do wonder what that means for performance (lifting the entire constant with the valtree -- valtrees are not interned so equality is probably expensive). Maybe valtree child lists should be interned to make this easier and faster?